common.blocks/version-selector/version-selector.js

```diff
@@ -20,15 +20,15 @@ modules.define('version-selector', [ 'i-bem__dom', 'querystring' ], function(pro
                         var ver = encodeURIComponent(qs.parse(location.href).version || '');
 
                         if (!ver) {
-                            ver = JSON.parse(select.dataset.bem);
-                            ver = ver['version-selector'].default.hash;
+                            ver = select.params.default.name;
                         }
 
                         select.setValue(ver);
                     };
 
                     select.domElem.on('change', function(e) {
-                        var val = e.target.value;
+                        var name = e.target.value;
+                        var hash = select._getHashByVersion(name);
```

Comment:
[Fixed this](https://github.com/bem/bem-xjst/pull/467/commits/7b5cab896c5111f5985999f467cdac31efa67a42)